### DISCUSSION TITLE
Database Connector Secrets

### DISCUSSION THREAD

**Discussion by finexus-howard** [airbytehq/airbyte#40229] [Connector Questions]
Hi,

I am new to Apache Airbyte. I have deployed an instance of Airbyte using docker compose and configured a source using the MySQL source connector in the dashboard. However, I have managed to see the password of my database is stored in the Airbyte database as plaintext, with the steps below:

1. "docker exec -ti airbyte-db psql -U docker -d airbyte"
2. SELECT * FROM secrets;

![Screenshot 2024-06-24 at 11 28 31 AM](https://github.com/airbytehq/airbyte/assets/107467916/9c0da7f4-c05b-4c35-aef4-ae9585f6b162)

May I know is there a way to store the database credentials in an encrypted form, or is there any other alternate best practise to configure Airbyte to store these secrets in a secure way?

Thank you.

**Comment by NAjustin** [comment]
@finexus-howard If you configure a Secrets Manager, these will be stored there . . . but you'll need to reconnect existing sources/destinations. The docs are light as this is a newer feature, but I believe that the supported Secret Managers are:
- Google Secret Manager
- AWS Secret Manager
- Hashicorp Vault

The most complete docs I've seen are probably in the [Platform config readme](https://github.com/airbytehq/airbyte-platform/tree/main/airbyte-config/config-secrets) in GitHub.

The Google option is mentioned in passing [here](https://docs.airbyte.com/operating-airbyte/security#credential-management), but you'll see other notes related to the supported options on the [original issue/feature request](https://github.com/airbytehq/airbyte/issues/837). Associated environment vars are noted [here](https://docs.airbyte.com/operator-guides/configuring-airbyte#secrets)

In Helm-based Kubernetes deployments, the `values.yaml` config for this feature is under `global.secretsManager`. This isn't mentioned much in the docs on the main site yet, but example config is shown for both [AWS](https://docs.airbyte.com/deploying-airbyte/on-kubernetes-via-helm#for-aws-s3-users) and [GCP](https://docs.airbyte.com/deploying-airbyte/on-kubernetes-via-helm#for-aws-s3-users) when it discusses storage.

[a comment by an automated account is not shown]
